[CPCN-1091] fix: Use 500 page size for Cards, Products, Topics, Companies & Users

MarkLark86 opened the pull request:
### Purpose
If there are 25 or more existing Topics for a single user, when creating more the front-end does not show these new Topics (as the default page size is 25).

### What has changed
~~Use [.search](https://github.com/superdesk/superdesk-core/blob/v3.0.0-alpha.3/superdesk/core/resources/service.py#L291) instead as the [.find](https://github.com/superdesk/superdesk-core/blob/v3.0.0-alpha.3/superdesk/core/resources/service.py#L696) method applies pagination. Also apply this change to other parts of the code that should use `search` instead~~

Use page size of 500 when getting resources like Cards, Products, Topics, Companies and Users (with filters applied).

**Note:** I reverted back to using `.find` as this applies the resource's configured default sort, where as `.search` does not. Applying a limit of 500 should be fine.

### Steps to test
Create 25 Topics, then create another one and check to see if they newer ones are available in the front-end.

![image](https://github.com/user-attachments/assets/9977bd9a-db94-489e-8b0c-a6e6cf666613)


Resolves: [CPCN-1091](https://sofab.atlassian.net/browse/CPCN-1091)


[CPCN-1091]: https://sofab.atlassian.net/browse/CPCN-1091?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MarkLark86 committed:
[CPCN-1091] fix: Use .search instead of .find due to auto pagination

MarkLark86 committed:
Revert "[CPCN-1091] fix: Use .search instead of .find due to auto pagination"

This reverts commit cfabb2145046786a857604018cd0b657b5d14444.

MarkLark86 committed:
Use 500 page size when getting resources

eos87 reviewed (approved):
Looks good. Quick thought: since we use the same for `CardsResourceService` in multiple places... would it worth it to put it in a separate service method and reuse it?

MarkLark86 committed:
Merge branch 'release/3.0' into CPCN-1091

MarkLark86 committed:
fix: Mgmt API - invalid handling of search by email

MarkLark86 committed:
improve: Set default page to 500 in base nhub service

MarkLark86 commented:
@eos87 I have adjusted the fix as per your suggestion. I changed the base Newshub async service to default to 500 page size (which works across system resources, excluding content/planning resources). This should be a cleaner solution.

This also raised a typing/bug in the ManagementAPI for users email search, so I fixed that one up too in the PR

eos87 reviewed (approved):
Neat solution!
